config/secrets/box/box.go

```diff
@@ -0,0 +1,87 @@
+// Package box is an asymmetric implementation of config/secrets using nacl/box
+package box
+
+import (
+	"github.com/micro/go-micro/v2/config/secrets"
+	"github.com/pkg/errors"
+	naclbox "golang.org/x/crypto/nacl/box"
+
+	"crypto/rand"
+)
+
+type box struct {
+	options secrets.Options
+
+	publicKey  [32]byte
+	privateKey [32]byte
+}
+
+// NewCodec returns a nacl-box codec
+func NewCodec(opts ...secrets.Option) secrets.Codec {
+	b := &box{}
+	for _, o := range opts {
+		o(&b.options)
+	}
+	return b
+}
+
+// Init initialises a box
+func (b *box) Init(opts ...secrets.Option) error {
+	for _, o := range opts {
+		o(&b.options)
+	}
+	if len(b.options.PrivateKey) != 32 || len(b.options.PublicKey) != 32 {
+		return errors.New("A 32 byte public and private key must be provided")
```

Comment:
i think that error must contains required len. Because i can provide both of them but not needed len.